bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -210,15 +212,33 @@ protected void resetExecutionContext(ScriptEngine engine, Map<String, ?> context
     protected @Nullable Object eval(ScriptEngine engine, String script) {
         try {
             if (compiledScript.isPresent()) {
+                if (engine instanceof Lock lock) {
+                    if (!lock.tryLock(1, TimeUnit.MINUTES)) {
+                        logger.error("Failed to acquire lock within one minute for script module of rule with UID '{}'",
+                                ruleUID);
+                        return null;
+                    }
+                }
                 logger.debug("Executing pre-compiled script of rule with UID '{}'", ruleUID);
-                return compiledScript.get().eval(engine.getContext());
+                Object result;
```

Comment:
Why do you create a new object here? You could also return it directly and move the `return null` from the `ScriptException` catch-block to the end of method (or return nul in the `finally` block.

Reply:
It "feels" better this way for me when reading the code, but I have changed it.